metrics-influxdb/src/main/java/com/izettle/metrics/influxdb/utils/InfluxDbWriteObjectSerializer.java

```diff
@@ -1,46 +1,94 @@
 package com.izettle.metrics.influxdb.utils;
 
-import java.io.IOException;
-import java.util.Map;
-
-import com.fasterxml.jackson.core.JsonGenerator;
-import com.fasterxml.jackson.core.Version;
-import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.izettle.metrics.influxdb.data.InfluxDbPoint;
 import com.izettle.metrics.influxdb.data.InfluxDbWriteObject;
-import com.fasterxml.jackson.databind.JsonSerializer;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.SerializerProvider;
+import java.text.NumberFormat;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
 
 public class InfluxDbWriteObjectSerializer {
 
-    protected static class MapSerializer<P, Q> extends JsonSerializer<Map<P, Q>> {
-        @Override
-        public void serialize(final Map<P, Q> influxDbMap, final JsonGenerator jsonGenerator, final SerializerProvider provider)
-                throws IOException {
-            if (influxDbMap != null) {
-                jsonGenerator.writeStartObject();
-                for (Map.Entry<P, Q> entry : influxDbMap.entrySet()) {
-                    jsonGenerator.writeFieldName(entry.getKey().toString());
-                    jsonGenerator.writeObject(entry.getValue());
-                }
-                jsonGenerator.writeEndObject();
-            }
+    // measurement[,tag=value,tag2=value2...] field=value[,field2=value2...] [unixnano]
+
+    /**
+     * calculate the lineprotocol for all Points.
+     *
+     * @return the String with newLines.
+     */
+    public String getLineProtocolString(InfluxDbWriteObject influxDbWriteObject) {
+        StringBuilder sb = new StringBuilder();
+        for (InfluxDbPoint point : influxDbWriteObject.getPoints()) {
+            sb.append(lineProtocol(point, influxDbWriteObject.getPrecision())).append("\n");
+        }
+        return sb.toString();
+    }
+
+    public String lineProtocol(InfluxDbPoint point, TimeUnit precision) {
```

Comment:
This should also return `StringBuilder`, alternatively you could add the `StringBuilder` as a parameter to avoid having to initialize a new one every time. If you want to keep the `String` return type for the public methods, then you should refactor it, so that the `StringBuilder` logic is in a private method which is used by `getLineProtocolString` and the public method just uses `toString`.